test/common/json/json_sanitizer_test_util.cc

```diff
@@ -148,6 +155,27 @@ bool utf8Equivalent(absl::string_view a, absl::string_view b, std::string& diffs
   }
 }
 
+bool jsonEquivalentStrings(absl::string_view sanitized, absl::string_view original,
+                           std::string& errmsg) {
+  for (uint32_t hex; !sanitized.empty() && !original.empty(); removePrefix(original, 1)) {
+    if (sanitized[0] == original[0]) {
```

Comment:
given a string with single back slash, `\`, our sanitize method should make ti to `\u005c`. They should be equal but in this test will return false?

Or I mis-understood this function?

Reply:
IMO, the logic here could be simpler. Because this is used to check if the sanitized result is same with original one. And every single byte that need escaping will be escaped

So, we can create a std::vector<unit32_t> to store all parsed code point. (the sanitized result should only contains characters that needn't escaping and \uxxxx). Then we can compare the std::vector<uint32_t> and the original string (they should have complete same length).

This maybe slower because we won't return early. But the readability will get a little improvement.

But this is test code, so, i am fine anyway as long as it could achieve our target.